packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -482,6 +513,38 @@ export class NodeProxyTarget extends ProxyTarget<Anchor> {
                 fail("`Forbidden` fields may not be deleted.");
         }
     }
+
+    public replaceField(fieldKey: FieldKey, newContent: ITreeCursor | ITreeCursor[]): void {
```

Comment:
The merge semantics are different for sequence and value/optional fields:
- For value/optional fields, the semantics are truly "replace" in the sense that if multiple clients concurrently call replace on a value/optional field, the last edit will win out and only the content in the last edit will remain in the field.
- For sequence fields, the semantics are a bit weird: if multiple clients concurrently call replace on a sequence field then all the insertions will be preserved. In addition to that, if you had nodes A and B in the sequence, and called replace with X Y, then an insert that targetted the gap between A and B may not end up between X and Y.

I think that's unfortunately the best we can offer right now.
In the future we can have both:
1. A "field replace" for sequences that doesn't preserve all insertions and doesn't line up X and Y with where A and B used to be.
2. A "content replace" for sequences that doesn't preserve all insertions and does line up X and Y with where A and B used to be.

Reply:
Note that this weirdness around sequence replacement won't affect JSON domains because they'll be using the `jsonArray` schema which will lead to the array node being replaced.

Reply:
Actually, the current API does not support the gaps in sequences, neither when inserting nor when replacing the nodes meaning that you can't actually target some arbitrary nodes - input data is always considered to be sequential. I have noted that in my list.

Reply:
My usage of the word "gap" here may be causing some confusion. I wasn't using it to mean something like having the sequence [A, undefined, B] (where index 1 could be thought of as a gap).
I just meant it as "an insert targeting the location between A and B in [A, B] such that one would expect to get [A, <interted-content>, B]".

Reply:
Thanks, I guess, I misunderstood your initial comment. Let's keep this discussion unresolved as a good reference for the merge semantics topic.